pkg/server/api/svid/v1/service.go

```diff
@@ -324,6 +328,81 @@ func (s *Service) NewJWTSVID(ctx context.Context, req *svid.NewJWTSVIDRequest) (
 	}, nil
 }
 
-func (s *Service) NewDownstreamX509CA(context.Context, *svid.NewDownstreamX509CARequest) (*svid.NewDownstreamX509CAResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+func (s *Service) NewDownstreamX509CA(ctx context.Context, req *svid.NewDownstreamX509CARequest) (*svid.NewDownstreamX509CAResponse, error) {
+	log := rpccontext.Logger(ctx)
+
+	if err := rpccontext.RateLimit(ctx, 1); err != nil {
+		log.WithError(err).Error("Rejecting request due to downstream CA signing rate limit")
+		return nil, err
+	}
+
+	downstreamEntries, isDownstream := rpccontext.CallerDownstreamEntries(ctx)
+	if !isDownstream {
+		err := status.Error(codes.NotFound, "error not found or not authorized")
+		log.WithError(err).Error("Rejecting NewDownstreamX509CA request because caller has no downstream entries")
+		return nil, err
+	}
+
+	entry := downstreamEntries[0]
+
+	spiffeID, err := api.IDFromProto(entry.SpiffeId)
+	if err != nil || spiffeID.String() != s.td.IDString() {
+		log.WithError(err).Error("Downstream entry has malformed SPIFFE ID")
+		return nil, status.Error(codes.Internal, "downstream entry has malformed SPIFFE ID")
+	}
```

Comment:
The SPIFFE ID on the downstream entry is not relevant to the SPIFFE ID of the downstream X.509 CA. We don't need this check.
```suggestion
```